src/coreclr/gc/gc.cpp

```diff
@@ -43608,35 +43663,6 @@ size_t gc_heap::desired_new_allocation (dynamic_data* dd,
                     new_allocation = min (new_allocation,
                                           max (min_gc_size, (max_size/3)));
                 }
-
-#ifdef DYNAMIC_HEAP_COUNT
```

Comment:
was this dead code?

Reply:
no, this was definitely used and the reason why it kept using only the min gen0 budget. I removed this and fixed it later in gc1